src/Datadog.Trace/Agent/Api.cs

```diff
@@ -195,6 +146,63 @@ private static HashSet<ulong> GetUniqueTraceIds(Span[][] traces)
             return uniqueTraceIds;
         }
 
+        private async Task<bool> SendTracesAsync(Span[][] traces, IApiRequest request)
+        {
+            IApiResponse response = null;
+
+            try
+            {
+                try
+                {
+                    _statsd?.AppendIncrementCount(TracerMetricNames.Api.Requests);
+                    response = await request.PostAsync(traces, _formatterResolver).ConfigureAwait(false);
+                }
+                catch
+                {
+                    // count the exceptions thrown by the HttpClient,
+                    // not responses with 5xx status codes
+                    // (which cause EnsureSuccessStatusCode() to throw below)
+                    _statsd?.AppendIncrementCount(TracerMetricNames.Api.Errors);
+                    throw;
+                }
+
+                if (_statsd != null)
+                {
+                    // don't bother creating the tags array if trace metrics are disabled
+                    string[] tags = { $"status:{response.StatusCode}" };
+
+                    // count every response, grouped by status code
+                    _statsd.AppendIncrementCount(TracerMetricNames.Api.Responses, tags: tags);
+                }
+
+                // Attempt a retry if the status code is not SUCCESS
+                if (response.StatusCode < 200 || response.StatusCode > 300)
```

Comment:
Only 2xx is success, not 300.

```suggestion
                if (response.StatusCode < 200 || response.StatusCode >= 300)
```

Reply:
You're right, but that's a copy/paste of the old code. Let's fix that in a separate PR